lua/cfc_voice_channels/core/sv/sv_player.lua

```diff
@@ -10,4 +10,21 @@ function player:isInChannel()
     end
 
     return false
-end
\ No newline at end of file
+end
+
+function player:getPlayerChannel()
+    for _, channel in pairs(cfc_voice.Channels) do
+        for _, ply in pairs(channel.Users) do
+            if self == ply then
+                return channel
+            end
+        end
+    end
+end
+
+hook.Add("PlayerDisconnected", "cfc_voice_on_disconnect", function(ply)
+    if not ply:isInChannel() then return end
+
+    local channel = ply:getPlayerChannel()
```

Comment:
I think this method name is too vague to add straight to the player table. `player:getPlayerVoiceChannel()` or even better, `player:getCFCVoiceChannel()`